osf/management/commands/import_EGAP.py

```diff
@@ -0,0 +1,186 @@
+    with open(os.path.join(egap_assets_path, epag_project_dir, 'project.json'), 'r') as fp:
+        project_data = json.load(fp)
+        title = project_data['title']
+        node = Node(title=title, creator=creator)
+        node.save()  # must save before adding contribs for auth reasons
+
+        for contributor in project_data['contributors']:
+            node.add_contributor_registered_or_not(
+                Auth(creator),
+                full_name=contributor['name'],
+                email=contributor['email'],
+                permissions=WRITE,
+                send_email='false'
+            )
+
+        node.set_visible(creator, visible=False, log=False, save=True)
+
+    return node
+
+
+def recursive_upload(auth, node, dir_path, parent='', metadata=list()):
+    try:
+        for item in os.listdir(dir_path):
+            item_path = os.path.join(dir_path, item)
+            base_url = '{}/v1/resources/{}/providers/osfstorage/{}'.format(WATERBUTLER_INTERNAL_URL, node._id, parent)
+            if os.path.isfile(item_path):
+                with open(item_path, 'rb') as fp:
+                    url = base_url + '?name={}&kind=file'.format(item)
+                    resp = requests.put(url, data=fp.read(), headers=auth)
+            else:
+                url = base_url + '?name={}&kind=folder'.format(item)
+                resp = requests.put(url, headers=auth)
+                metadata = recursive_upload(auth, node, item_path, parent=resp.json()['data']['attributes']['path'], metadata=metadata)
+
+            if resp.status_code == 409:  # if we retry something already uploaded just skip.
+                continue
+
+            if resp.status_code != 201:
+                raise EGAPUploadException('Error waterbutler response is {}, with {}'.format(resp.status_code, resp.content))
+
+            metadata.append(resp.json())
+    except EGAPUploadException as e:
+        logger.info(str(e))
+        metadata = recursive_upload(auth, node, dir_path, parent=parent, metadata=metadata)
+
+    return metadata
+
+
+def get_egap_assets(guid, creator_auth):
+    node = Node.load(guid)
+    zip_file = node.files.first()
+    temp_path = tempfile.mkdtemp()
+
+    url = '{}/v1/resources/{}/providers/osfstorage/{}'.format(WATERBUTLER_INTERNAL_URL, guid, zip_file._id)
+    zip_file = requests.get(url, headers=creator_auth).content
+
+    egap_assets_path = os.path.join(temp_path, 'egap_assets.zip')
+
+    with open(egap_assets_path, 'w') as fp:
+        fp.write(zip_file)
+
+    with ZipFile(egap_assets_path, 'r') as zipObj:
+        zipObj.extractall(temp_path)
+
+    return temp_path
+
+
+def main(guid, creator_username):
+    egap_schema = ensure_egap_schema()
+    creator, creator_auth = get_creator_auth_header(creator_username)
+
+    egap_assets_path = get_egap_assets(guid, creator_auth)
+
+    directory_list = [directory for directory in os.listdir(egap_assets_path) if directory not in ('egap_assets.zip', '__MACOSX')]
```

Comment:
Probably a good idea to mention what the `__MACOSX` file is and why it's being excluded.